plugins/algoltek-usb/fu-algoltek-usb-device.c

```diff
@@ -0,0 +1,537 @@
+/*
+ * Copyright (C) 2023 Algoltek, Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-algoltek-usb-common.h"
+#include "fu-algoltek-usb-device.h"
+#include "fu-algoltek-usb-firmware.h"
+#include "fu-algoltek-usb-struct.h"
+
+struct _FuAlgoltekUsbDevice {
+	FuUsbDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuAlgoltekUsbDevice, fu_algoltek_usb_device, FU_TYPE_USB_DEVICE)
+
+static gboolean
+algoltek_device_ctrl_transfer(FuAlgoltekUsbDevice *self,
+			      GUsbDeviceDirection direction,
+			      guint8 algoltek_cmd,
+			      guint16 value,
+			      guint16 index,
+			      GByteArray *transfer_pkt,
+			      guint8 len,
+			      GError **error)
+{
+	GUsbDevice *usb_device = fu_usb_device_get_dev(FU_USB_DEVICE(self));
+
+	if (!g_usb_device_control_transfer(usb_device,
+					   direction,
+					   G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					   G_USB_DEVICE_RECIPIENT_INTERFACE,
+					   algoltek_cmd,
+					   value,
+					   index,
+					   transfer_pkt->data,
+					   len,
+					   NULL,
+					   ALGOLTEK_DEVICE_USB_TIMEOUT,
+					   NULL,
+					   error)) {
+		g_prefix_error(error, "Device communication failure: ");
```

Comment:
normally with `g_prefix_error` it's all lowercase, i.e. `device communication failure: ` -- there are a few like this.

Reply:
Updated!